code/datums/emotes.dm

```diff
@@ -81,6 +82,21 @@
 	else
 		user.visible_message(msg)
 
+//austation begin -- cooldown proc bb
+
+/// For handling emote cooldown, return true to allow the emote to happen
+/datum/emote/proc/check_cooldown(mob/user, intentional)
+	if(!intentional)
+		return TRUE
+	if(user.emotes_used && user.emotes_used[src] + cooldown > world.time)
+		return FALSE
+	if(!user.emotes_used)
+		user.emotes_used = list()
+	user.emotes_used[src] = world.time
+	return TRUE
+
+//austation end
+
```

Comment:
This proc can also be defined in a modularized emotes.dm file, since it's brand new and not a change to an existing proc. Just stick this whole chunk of code into the file.